### DISCUSSION TITLE
Oqtane module in server only mode with EF core

### DISCUSSION THREAD

**Discussion by avazon** [oqtane/oqtane.framework#2755] [Q&A]
Is it possible to create modules only för Blazor Server and uses EF core direct access? I work with LOB applications small but many registers/tables. The template created from Oqtane module creater is for both Wasm and server with api's for data access. In my case there are a lot of apis for all registers to create. In a server app I can use sql access direct which is preferred for me with data only on the server and quicker response for queries.
Any ideas how I could create "Blazor server only" modules with direct database sql access and still use the tools/classes in Oqtane for multitenant dbcontext's? My guess it is in conflict with the design of Oqtane unfortunately.

**Comment by lanthonyneville** [comment]
I had a similar question - see here: [Viability of building modules as single assemblies (rather than Client + Server + Shared) for simple cases](https://github.com/oqtane/oqtane.framework/discussions/2594)

**Reply by avazon** [reply]
Thank you for your input lanthonyneville.  I will try to do similar.

**Comment by sbwalker** [comment]
There should be no issue in creating an Oqtane module which is intended to only function in Blazor Server environments. You could create a solution with a single project which contains your razor components, repositories, and models (in this approach you would not need multiple projects, Services, or Controllers). You should be able to set up your module's DBContext using the standard DBContextBase which will support the multi-tenancy capabilities. Have you tried this and run into difficulties?

**Reply by avazon** [reply]
Thank you for your input. I have not tried it and run into difficultied. It is promising you say it should work.
